object.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2021 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nats
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/nats-io/nuid"
+)
+
+type ObjectStore interface {
```

Comment:
It would seem that all KVs are a simplification of the ObjectStore, the blob storage is simply saying that the payload is larger than it fits in some message and some checksum is applied to validate against corruption. Wondering if merging the concepts has any value.

Reply:
I have a few toy apps that combine them already I was playing with over the weekend.